sdk/ml/azure-ai-ml/azure/ai/ml/_schema/pipeline/pipeline_component.py

```diff
@@ -107,21 +107,18 @@ def _post_load_pipeline_jobs(context, data: dict) -> dict:
                 job_instance = AutoMLJob._create_instance_from_schema_dict(
                     loaded_data=job_instance,
                 )
-                jobs[key] = job_instance
             elif job_instance.get("type") == ControlFlowType.IF_ELSE:
                 # Convert to if-else node.
                 job_instance = ConditionNode._create_instance_from_schema_dict(loaded_data=job_instance)
-                jobs[key] = job_instance
             elif job_instance.get("type") == ControlFlowType.DO_WHILE:
                 # Convert to do-while node.
                 job_instance = DoWhile._create_instance_from_schema_dict(pipeline_jobs=jobs, loaded_data=job_instance)
-                jobs[key] = job_instance
             elif job_instance.get("type") == ControlFlowType.PARALLEL_FOR:
                 # Convert to do-while node.
                 job_instance = ParallelFor._create_instance_from_schema_dict(
                     pipeline_jobs=jobs, loaded_data=job_instance
                 )
-                jobs[key] = job_instance
+            jobs[key] = job_instance
```

Comment:
why this change can refine the CLI error message?

Reply:
This is not related to error message, just to fix your previous comment. Forgot to mention you here.